### PR TITLE
EDE-1338 Pass user first name to HTML xblock

### PR DESCRIPTION
**Pull request by zahid-arbisoft**
### [EDE-1338](https://edlyio.atlassian.net/browse/EDE-1338) Pass User first name to HTML xblock

### CODE REVIEW

File: common/djangoapps/xblock_django/user_service.py [hunk 1]
```diff
@@ -81,7 +81,11 @@ def _convert_django_user_to_xblock_user(self, django_user):
             # need this check because profile is only created through registration flow
             full_name = full_name if full_name is not None and full_name != '' else "{} {}".format(
                 django_user.first_name, django_user.last_name)
+            first_name = django_user.first_name
+            # extract first name from full name, if profile does not have first name explicitly
+            first_name = first_name if first_name is not None and first_name != '' else full_name.split()[0]
```
Comment (emzubair) [inline common/djangoapps/xblock_django/user_service.py:86]: We can rely on `full_name` to extract `first_name` but we should check for a blank full name.
ps please `title()` the full name as well.

Comment (emzubair) [reply common/djangoapps/xblock_django/user_service.py:86]: `django_user.first_name` will return blank instead of `None`

**Review by emzubair — CHANGES_REQUESTED** [review]

File: common/djangoapps/xblock_django/user_service.py (hunk 1 above)
Comment (zahid-arbisoft) [reply common/djangoapps/xblock_django/user_service.py]: I have updated the code

File: common/djangoapps/xblock_django/user_service.py [hunk 2]
```diff
@@ -81,7 +81,14 @@ def _convert_django_user_to_xblock_user(self, django_user):
             # need this check because profile is only created through registration flow
             full_name = full_name if full_name is not None and full_name != '' else "{} {}".format(
                 django_user.first_name, django_user.last_name)
+            full_name = full_name.title()
+
+            first_name = ''
+            if full_name.strip() != '':
```
Comment (emzubair) [inline common/djangoapps/xblock_django/user_service.py:87]: @zahid-arbisoft  not necessary but pythonic version looks good :) i.e
`first_name  = '' if full_name.strip() == '' else full_name.split()[0]`

**Review by emzubair — DISMISSED** [review]

File: common/djangoapps/xblock_django/user_service.py (hunk 2 above)
Comment (zahid-arbisoft) [reply common/djangoapps/xblock_django/user_service.py]: Nice catch, I forgot to do it. I have updated the code.

**Review by emzubair — APPROVED** [review]

**Pull request merged**